src/newwave/entity.mo

```diff
@@ -17,88 +17,275 @@ import Result "mo:base/Result";
+   *
+   * @return A new entity with the new fromIds field
+  */
+  public func updateEntityFromIds(entity : Entity, fromIds : [Text]) : Entity {
+    return {
+      id = entity.id;
+      creationTimestamp = entity.creationTimestamp;
+      creator = entity.creator;
+      owner = entity.owner;
+      settings = entity.settings;
+      entityType = entity.entityType;
+      name = entity.name;
+      description = entity.description;
+      keywords = entity.keywords;
+      entitySpecificFields = entity.entitySpecificFields;
+      listOfEntitySpecificFieldKeys = entity.listOfEntitySpecificFieldKeys;
+      fromIds = fromIds;
+      toIds = entity.toIds;
+    };
+  };
+
+  /**
+   * Function updates an entity and returns a new entity with the toIds updated to the new value
+   *
+   * @return A new entity with the new toIds field
+  */
+  public func updateEntityToIds(entity : Entity, toIds : [Text]) : Entity {
+    return {
+      id = entity.id;
+      creationTimestamp = entity.creationTimestamp;
+      creator = entity.creator;
+      owner = entity.owner;
+      settings = entity.settings;
+      entityType = entity.entityType;
+      name = entity.name;
+      description = entity.description;
+      keywords = entity.keywords;
+      entitySpecificFields = entity.entitySpecificFields;
+      listOfEntitySpecificFieldKeys = entity.listOfEntitySpecificFieldKeys;
+      fromIds = entity.fromIds;
+      toIds = toIds;
+    };
+  };
+
+  /**
+   * Function takes a entity update object and an already created entity and updates the entity with the new values
+   * provided by the entity update object
+   *
+   * @return The new entity with the values updated with the entity update values
+  */
+  public func updateEntityFromUpdateObject(entityUpdateObject : EntityUpdateObject, originalEntity : Entity) : Entity {
+    return {
+      id = originalEntity.id;
+      creationTimestamp = originalEntity.creationTimestamp;
+      creator = originalEntity.creator;
+      owner = originalEntity.owner;
+      settings = Option.get<EntitySettings>(entityUpdateObject.settings, originalEntity.settings);
+      entityType = originalEntity.entityType;
+      name = Option.get<?Text>(?entityUpdateObject.name, originalEntity.name);
+      description : ?Text = Option.get<?Text>(?entityUpdateObject.description, originalEntity.description);
+      keywords : ?[Text] = Option.get<?[Text]>(?entityUpdateObject.keywords, originalEntity.keywords);
+      entitySpecificFields = originalEntity.entitySpecificFields;
+      listOfEntitySpecificFieldKeys = originalEntity.listOfEntitySpecificFieldKeys;
+      fromIds = originalEntity.fromIds;
+      toIds = originalEntity.toIds;
+    };
   };
 
+  /**
+   This type defines the fields that the current owner is allowed
+   to modify and use to update the entity
+  */
   public type EntityUpdateObject = {
-    internalId : Text;
-    settings : ?EntitySettings.EntitySettings;
+    /**
+     * The ID of the entity to update
```

Comment:
maybe add a note that this field is for identification only and won't be updated